src/app/layout.tsx

```diff
@@ -8,6 +8,8 @@ import { cn } from '@/lib/tailwind/utils';
 import { Metadata } from 'next';
 import { ReactNode } from 'react';
 
```

Comment:
We have to use `dynamic rendering` to show the different contents for each user by using Cookies.
[Amplify doc](https://docs.amplify.aws/react/build-a-backend/server-side-rendering/nextjs/#with-nextjs-app-router)

Reply:
I fixed scope of `dynamic rendering` from overall `app/layout.tsx` to each page.tsx with this [commit](https://github.com/genesis-tech-tribe/nishiki-frontend/pull/155/commits/b8606356c96e17b2b8dc081aa1756da144eaa9b5).

Reply:
@kotaaaa 
https://nextjs.org/docs/app/building-your-application/caching#opting-out-1

I read the document on dynamic data fetching, and my understanding is that if we use the `fetch()` function, the first option is to specify `{ cache: 'no-store' }` as an argument. Otherwise, we can use `export const dynamic = 'force-dynamic'`. However, your approach is totally acceptable and works well.

What are your thoughts on this?

Reply:
Thank you for your comment!
I checked the situation with setting `{ cache: 'no-store' }` in `fetch()` options and not setting `export const dynamic = 'force-dynamic';` in page level, but it turns out same build error happened while generating static page.

```
Generating static pages (0/8)  [    ]Error: API response is invalid
```

So that means I understand that we have to set `export const dynamic = 'force-dynamic';` clearly for now when we want to use this [Amplify library for Dynamic Rendering](https://docs.amplify.aws/react/build-a-backend/server-side-rendering/nextjs/#with-nextjs-app-router), so I will make it leave.

PS, it is not directly related, but I noticed that I should improve flexibility for cache strategy for `fetch()` method, so I fixed common method a little.

Reply:
Understood. Thank you for checking it :)